### PR TITLE
How to: keep track of token counts

### PR DESCRIPTION
**Pull request by eyurtsev**
* Update LLM keep track of token counts
* Warning about streaming with OpenAI 
* Explain that the callbacks are only for OpenAI
* Add link to langsmith since it provides a more generic solution

### CODE REVIEW

File: docs/versioned_docs/version-0.2.x/how_to/llm_token_usage_tracking.ipynb [hunk 1]
```diff
@@ -2,169 +2,228 @@
...
+    "\n",
+    "## Using LangSmith\n",
+    "\n",
+    "You can use [LangSmith](https://www.langchain.com/langsmith) to help track token usage in your LLM application. See the [LangSmith quick start guide](https://docs.smith.langchain.com/).\n",
+    "\n",
+    "## Using callbacks\n",
+    "\n",
+    "There are some API-specific callback context managers that allow you to track token usage across multiple calls. You'll need to check whether such an integration is available for your particular model.\n",
+    "\n",
+    "If such an integration is not available for your model, you can create a custom callback manager by adapting the implementation of the OpenAI callback manager (find it in the code base and adapt for your own LLMs.)\n",
```
Comment (ccurme) [inline docs/versioned_docs/version-0.2.x/how_to/llm_token_usage_tracking.ipynb:28]:
```suggestion
    "If such an integration is not available for your model, you can create a custom callback manager by adapting the implementation of the [OpenAI callback manager](https://api.python.langchain.com/en/latest/_modules/langchain_community/callbacks/openai_info.html#OpenAICallbackHandler).\n",
```

File: docs/versioned_docs/version-0.2.x/how_to/llm_token_usage_tracking.ipynb [hunk 2]
```diff
@@ -2,169 +2,228 @@
...
+    "token counts when streaming.\n",
     "\n",
-    "llm = OpenAI(temperature=0)\n",
-    "tools = load_tools([\"serpapi\", \"llm-math\"], llm=llm)\n",
-    "agent = initialize_agent(\n",
-    "    tools, llm, agent=AgentType.ZERO_SHOT_REACT_DESCRIPTION, verbose=True\n",
-    ")"
+    "If you want to count token counts correctly, for streaming, either\n",
+    "implement a custom callback handler that uses an appropriate tokenizers to\n",
+    "count the tokens or use a monitoring platform like [LangSmith](https://www.langchain.com/langsmith).\n",
```
Comment (ccurme) [inline docs/versioned_docs/version-0.2.x/how_to/llm_token_usage_tracking.ipynb:175]: Consider adding something like "Note that below no tokens are counted by the callback handler, because this metadata is not currently supplied by OpenAI in streaming mode."

File: docs/versioned_docs/version-0.2.x/how_to/llm_token_usage_tracking.ipynb (hunk 1 above)
Comment (sepiatone) [reply docs/versioned_docs/version-0.2.x/how_to/llm_token_usage_tracking.ipynb:28]: @ccurme Did you mean to link to the the OpenAI callback handler `OpenAiCallbackHandler` or the OpenAI specific context manager `get_openai_callback()`?

Comment (ccurme) [reply docs/versioned_docs/version-0.2.x/how_to/llm_token_usage_tracking.ipynb:28]: Yeah good call that fits the context better: https://api.python.langchain.com/en/latest/_modules/langchain_community/callbacks/manager.html#get_openai_callback

File: docs/versioned_docs/version-0.2.x/how_to/llm_token_usage_tracking.ipynb [hunk 3]
```diff
@@ -2,169 +2,228 @@
...
+    "\n",
+    "If such an integration is not available for your model, you can create a custom callback manager by adapting the implementation of the OpenAI callback manager (find it in the code base and adapt for your own LLMs.)\n",
+    "\n",
+    "### OpenAI\n",
+    "\n",
+    "Let's first look at an extremely simple example of tracking token usage for a single Chat model call.\n",
+    "\n",
+    ":::{.callout-danger}\n",
+    "\n",
+    "The callback handler does **NOT** work when **streaming** becauyse OpenAI does not return token information in its API response.\n",
```
Comment (sepiatone) [inline docs/versioned_docs/version-0.2.x/how_to/llm_token_usage_tracking.ipynb:36]: @eyurtsev Related - https://community.openai.com/t/usage-stats-now-available-when-using-streaming-with-the-chat-completions-api-or-completions-api/738156

tl;dr Set stream_options: `{"include_usage": true}` in the request and an additional chunk will be streamed at the end that will contain the `usage` data.

Comment (ccurme) [reply docs/versioned_docs/version-0.2.x/how_to/llm_token_usage_tracking.ipynb:36]: Good callout. Need to be careful with how we do this but I put up a PR here: https://github.com/langchain-ai/langchain/pull/21548
